packages/mongodb-memory-server-core/src/MongoMemoryReplSet.ts

```diff
@@ -261,28 +261,14 @@ export default class MongoMemoryReplSet extends EventEmitter {
     return server;
   }
 
-  async _waitForPrimary(timeout: number = 30000): Promise<boolean> {
-    if (!this.admin) {
-      return false;
-    }
-    const replStatus: ReplStatusResultT = await this.admin.command({
-      serverStatus: 1,
-      metrics: 0,
-      locks: 0,
-    });
-    this.debug('   replStatus:', replStatus);
-    const hasPrimary = replStatus.repl.ismaster;
-    if (!hasPrimary) {
-      const restTimeout = timeout - 500;
-      if (restTimeout <= 0) {
-        throw new Error(`No PRIMARY elected yet. Timeout expired. Exiting...`);
-      }
-      this.debug(`No PRIMARY yet. Waiting... ${restTimeout}ms`);
-      return new Promise((resolve) =>
-        setTimeout(() => resolve(this._waitForPrimary(restTimeout)), 500)
-      );
-    }
+  async _waitForPrimary(): Promise<void> {
+    await Promise.race(
+      this.servers.map((server) => {
+        const instanceInfo: any = server.getInstanceInfo();
+        return instanceInfo.instance.waitPrimaryReady();
+      })
+    );
 
```

Comment:
@jardakotesovec How do you handle the error cases here ? For exemple if no primary instance is detected, what happens ? We don't have any trace as well for this kind of scenario.

And at both places where  `_waitForPrimary()` is called there is no try catch wrapping it as well :/ 
https://github.com/nodkz/mongodb-memory-server/blob/f3c5964e4c4954386f7c70422ea1add4946498ef/packages/mongodb-memory-server-core/src/MongoMemoryReplSet.ts#L136-L152
https://github.com/nodkz/mongodb-memory-server/blob/f3c5964e4c4954386f7c70422ea1add4946498ef/packages/mongodb-memory-server-core/src/MongoMemoryReplSet.ts#L210-L250

Maybe should we improve error handling for this function by throwing an error in it and logging this error in try catch wrappers where `_waitForPrimary` is called, what do you think about this improvement ?

Reply:
Only way to handle failure is some overall timeout when waiting for primary - based on what you would want to throw error?

That was no failure scenario before my PRs for _waitForPrimary, it was waiting indefinitely for some condition.  @nodkz introduced timeout for this function as quick workaround because  my previous PR  did not work reliably on the CI. Hence I came up with this implementation which I believe will be reliable. 

I am opened to also include the timeout there as failure scenario, but priority is to make it reliable :-).

Reply:
@AJRdev added new commit with timeout logic as failure scenario. Similar to waht @nodkz added as hotfix. That won't hurt.

Reply:
Ok I wasn't aware the error handler before was juste a quick workaround for the CI, well this should be good for the moment :-)